### DISCUSSION TITLE
OpenVDB Core Library Hard dependency on Boost

### DISCUSSION THREAD

**Discussion by jhenriques** [AcademySoftwareFoundation/openvdb#1562] [Q&A]
Hi all,

I'm trying to build a minimal version of OpenVDB. I configured it to not include boost. Here is the cmake line:

`cmake .. -GXcode -DOPENVDB_BUILD_BINARIES=OFF -DOPENVDB_CORE_SHARED=OFF -DOPENVDB_USE_DELAYED_LOADING=OFF -DUSE_STATIC_DEPENDENCIES=ON -DBUILD_SHARED_LIBS=OFF`

But then I get a build error here:

https://github.com/AcademySoftwareFoundation/openvdb/blob/2caed723981eb16799f825aeb35a1fd93e541383/openvdb/openvdb/math/Math.h#L13

So, I guess the current version for OpenVDB Core Library actually depends on boost but it's not stated anywhere?

**Comment by Idclip** [comment]
Hi @jhenriques,

We are working on stripping out the boost dependency entirely, but yes it is still required (as of VDB 10.X). We have our dependency requirements listed here:

https://www.openvdb.org/documentation/doxygen/dependencies.html

It looks like this has been incorrectly updated to imply that boost is optional for the core library. I'll fix this documentation.

edit: Ah, looks like theres a deeper issue here. The updates to VDB 10's CMake to disable delay loading incorrectly remove the entirety of the boost dependency. @danrbailey this looks like a mistake? I'm not sure how the CI is passing, maybe the system boost is being picked up

**Reply by Idclip** [reply]
https://github.com/AcademySoftwareFoundation/openvdb/pull/1565 fixes our build system and documentation (which we will re publish)

**Comment by bpmckinnon** [comment]
Hello, I wanted to check on the progress of this one.  I've done a test with std::any vs boost::any and they seem to have a 1:1 mapping and I was able to replace them without issue (limited testing).  It seems like a reasonable change too given that OpenVDB is minimum c++17.  The conversion_traits is confounding me a bit, but it looks like it's the last remaining required link to boost.

**Comment by jdumas** [comment]
Hi. It's been a while since this discussion opened, but I was wondering if removing the boost dependency is still something of interest @Idclip @danrbailey ?

**Comment by Idclip** [comment]
Hi @jdumas, yeah I think so. Here's where I believe we stand:

 -  It's been a minute, but I believe boost is now truly an optional dependency (only required with OPENVDB_USE_DELAYED_LOADING=ON, the default) which was our main goal. It would still be cool to remove it completely.
 - The delay loading stuff is still a heavily used feature so I doubt its turned off much.
 - I am a bit wary of the amount of work required to remove boost iostreams/interprocess. It didn't look particularly trivial and the obvious alternative (us having a native solution) isn't ideal if that solution is large.
 - There are ongoing conversations around some file format work which may impact how useful the whole delay loading system may turn out to be, though this is still a pretty theoretical roadmap item. 

@danrbailey any thoughts?
